pytket/extensions/qiskit/backends/crosstalk_model.py

```diff
@@ -0,0 +1,378 @@
+from typing import List, Tuple, Union, Dict
+from dataclasses import dataclass
+import math
+
+from pytket.circuit import (
+    Circuit,
+    Qubit,
+    Node,
+    Command,
+    OpType,
+    Op,
+    Unitary1qBox,
+    Unitary2qBox,
+    Unitary3qBox,
+)
+from pytket.backends.backendinfo import BackendInfo
+from pytket.extensions.qiskit.qiskit_convert import _gate_str_2_optype
+
+import numpy as np
+
+
+@dataclass
+class FractionalUnitary:
+    """
+    Wrapper for a fractional unitary gate
+    :param cmd: the fractional UnitaryBox wrapped in a pytket Command
+    :param n_fractions: the number of fractional gates used to compose the original unitary gate.
+
+    """
+
+    cmd: Command
+    n_fractions: float
+
+
+@dataclass
+class NoiseGate:
+    """
+    Wrapper for a gate that simulating noise
+    :param cmd: gate wrapped in a pytket Command
+    :param type: one of zz_crosstalks, single_q_phase, two_q_induced_phase
+    and non_markovian.
+    """
+
+    cmd: Command
+    type: str
+
+
+Instruction = Union[FractionalUnitary, Command, NoiseGate]
+Slice = List[Instruction]
+EPS = 1e-9
+
+
+@dataclass
+class CrosstalkParams:
+    """
+    Stores various parameters for modelling crosstalk noise
+
+    :param zz_crosstalks: symmetric crosstalks between qubit pairs
+    :param single_q_phase_errors: dict specify the single qubit phase error on each qubit
+    :param two_q_induced_phase_errors: keys of dictionary specify the control and target qubit index,
+    while the values are tuples with the spectator qubit index and the amount of phase error to be applied.
+    :param non_markovian_noise: List storing the non-Markovian noise parameters.
+    Each tuple in the list contains the qubit index and the zx, zz noise parameters.
+    :param virtual_z: If True, then don't break any single qubit Z gate into unitary fractions,
+    instead add the full unitary.
+    :param N: hyperparameter N
```

Comment:
On a first pass it's not clear from this docstring how I should be choosing N. Is there a way to clarify this here?